src/hooks/useSingIn.ts

```diff
@@ -0,0 +1,26 @@
+import { useMutation } from "@tanstack/react-query";
+import { UserAuthentication } from "../types/userAuthentication";
+import { useNavigate } from "react-router-dom";
+
+import { singIn } from "../services/auth";
+import { SuccesfulSingIn } from "../types/succesfulSingIn";
+
+
+const useSignIn = () => {
+  const navigate = useNavigate();
+  return useMutation<SuccesfulSingIn, Error, UserAuthentication>({
+    mutationFn: singIn,
+    onSuccess: (newUser, variables) => {
+      localStorage.setItem("token", newUser.idToken);
+     
+      if (variables.email.includes("admin")) {
```

Comment:
extract the logic into a function -> isUserAdmin()

So we can easily change it in the future